python/clx/analytics/sequence_classifier.py

```diff
@@ -0,0 +1,351 @@
+
+        if torch.cuda.is_available():
+            self._device = torch.device("cuda")
+            self._model.cuda()
+            self._model = nn.DataParallel(self._model)
+        else:
+            self._device = torch.device("cpu")
+
+    def train_model(
+        self,
+        train_data,
+        labels,
+        learning_rate=3e-5,
+        max_seq_len=128,
+        batch_size=32,
+        epochs=5,
+    ):
+        """
+        Train the classifier
+
+        :param train_data: text data for training
+        :type train_data: cudf.Series
+        :param labels: labels for each element in train_data
+        :type labels: cudf.Series
+        :param learning_rate: learning rate
+        :type learning_rate: float
+        :param max_seq_len: Limits the length of the sequence returned by tokenizer. If tokenized sentence is shorter than max_seq_len, output will be padded with 0s. If the tokenized sentence is longer than max_seq_len it will be truncated to max_seq_len.
+        :type max_seq_len: int
+        :param batch_size: batch size
+        :type batch_size: int
+        :param epoch: epoch, default is 5
+        :type epoch: int
+
+        Examples
+        --------
+        >>> from cuml.preprocessing.model_selection import train_test_split
+        >>> emails_train, emails_test, labels_train, labels_test = train_test_split(train_emails_df, 'label', train_size=0.8)
+        >>> sc.train_model(emails_train, labels_train)
+        """
+        train_gdf = cudf.DataFrame()
+        train_gdf["text"] = train_data
+        train_gdf["label"] = labels
+        (
+            train_data,
+            validation_data,
+            train_labels,
+            validation_labels,
+        ) = train_test_split(train_gdf, "label", train_size=0.8, random_state=2)
+
+        train_inputs, train_masks = self._bert_uncased_tokenize(train_data["text"], max_seq_len)
+        validation_inputs, validation_masks = self._bert_uncased_tokenize(validation_data["text"], max_seq_len)
+
+        # convert labels to tensors
+        train_labels = torch.tensor(train_labels.to_array())
+        validation_labels = torch.tensor(validation_labels.to_array())
+
+        # create dataloaders
+        train_dataset = TensorDataset(train_inputs, train_masks, train_labels)
+        train_sampler = RandomSampler(train_dataset)
+        train_dataloader = DataLoader(train_dataset, sampler=train_sampler, batch_size=batch_size)
+
+        validation_dataset = TensorDataset(validation_inputs, validation_masks, validation_labels)
+        validation_sampler = SequentialSampler(validation_dataset)
+        validation_dataloader = DataLoader(validation_dataset, sampler=validation_sampler, batch_size=batch_size)
+
+        self._config_optimizer(learning_rate)
+
+        self._model = self._train(train_dataloader, validation_dataloader, self._model, epochs)
+
+        torch.cuda.empty_cache()
+        gc.collect()
```

Comment:
```
torch.cuda.empty_cache()
gc.collect()
```
Does this affect only variables within this scope? I might double check on using these commands in source code. Have we tried marking variables for deletion using `del`?

Reply:
thanks @brhodes10. those were added to try addressing oom issues i had earlier. you're right, lines are not necessary so i'm removing them.